server_addon/blender/client/ayon_blender/plugins/publish/extract_fbx.py

```diff
@@ -55,8 +55,9 @@ def process(self, instance):
                 new_materials.append(mat)
                 new_materials_objs.append(obj)
 
-        scale_length = bpy.context.scene.unit_settings.scale_length
-        bpy.context.scene.unit_settings.scale_length = 0.01
+        scene = bpy.context.scene
+        scale_length = scene.unit_settings.scale_length
+        scene.unit_settings.scale_length = instance.data.get("unitScale")
```

Comment:
Shouldn't this just use the `global_scale` argument of the export function? [see here](https://docs.blender.org/api/current/bpy.ops.export_scene.html#module-bpy.ops.export_scene) or why is changing the scene better?

If we are to change the scene - then this should be done via a context manager!

Reply:
agreed. This should be done with the contextmanager + context.exitstack()
You can get some inspirations here: https://github.com/ynput/ayon-core/pull/674#discussion_r1649068696